kb/development/script-coding-standards.md

```diff
@@ -0,0 +1,102 @@
+
+```python
+import scripts_path # pylint: disable=unused-import
+from mbedtls_dev import …
+```
+
+### Python imports
+
+We follow the usual convention that imports go at the top of a file (after the docstring and the license comment). Imports are typically ordered as follows:
+
+1. Standard library, in lexicographic order.
+2. Third-party libraries, if any, in lexicographic order.
+3. Modules from `mbedtls_dev`, in lexicographic order.
+
+Avoid importing individual names from modules, as it can be difficult to figure out which module an unqualified name comes from. It's ok to import names if they're sufficiently recognizable, for example:
+
+* Importing individual modules from `mbedtls_dev` is fine.
+* Importing capitalized type names from `typing` is fine.
+* Importing individual submodules from Cryptodome is fine.
+
+### Third-party Python libraries
+
+Avoid bringing in new dependencies, because each dependency adds risk to the project. Only use dependencies if they have a clear advantage.
+
+The minimum supported version of third-party Python libraries is recorded in `*.requirements.txt` files in the [`scripts` directory](https://github.com/Mbed-TLS/mbedtls/tree/development/scripts). As with the [Python version](#language-versions), keep in mind that long-time support branches may need to support older versions.
+
+### Python language features
+
+The maintainers of Mbed TLS are primarily C programmers, not Python programmers. It's ok to use fancy language features if they're useful, but don't necessarily assume that the reader is familiar with them. Make sure that if a reader is unfamiliar with an advanced feature, they'll be able to easily look it up.
+
+* Classes, methods and simple inheritance can be used freely.
+* Avoid metaclasses and nontrivial dynamic class creation. If you use these features, write documentation that's accessible to non-experts.
+* Do define [special methods](https://docs.python.org/3/reference/datamodel.html#special-method-names) if convenient for comparison, context managers, etc. But avoid customizing attribute access.
+* Use [decorators](https://docs.python.org/3/glossary.html#term-decorator) if they have a clear benefit.
+* Do use [generators](https://docs.python.org/3/glossary.html#term-generator) and [list comprehension](https://docs.python.org/3/glossary.html#term-list-comprehension) where warranted.
+
+### Python typing
+
+Using type annotations is encouraged. We run [mypy](http://mypy-lang.org/) to verify static typing.
+
+Avoid complex overloading. It's generally easier to understand code if it only expects a specific type of data.
+
+### Python style conventions
+
+We generally follow [PEP 8](https://peps.python.org/pep-0008/), and [PEP 257](https://peps.python.org/pep-0257/) for docstrings.
```

Comment:
As mentioned above, we could refer to pep for anything not covered not only docstrings.

Reply:
Uh, that's exactly what this sentence does?

Reply:
I think this is one of those cases where it would be nice if English grammar had parentheses. What is meant is "we generally follow (PEP 8) and (PEP 257 for docstrings)", but could be read as "we generally follow (PEP 8 and PEP 257) for docstrings".

This could perhaps be reworded to "we generally follow PEP 8 for code and PEP 257 for docstrings" for the avoidance of doubt, but it seems a minor thing.

Reply:
Oh, I see. The comma actually makes it unambiguous, but it's subtle. I've reformulated.